cmd/lakectl/cmd/ingest.go

```diff
@@ -12,6 +14,32 @@ const ingestSummaryTemplate = `
 Staged {{ .Objects | yellow }} external objects (total of {{ .Bytes | human_bytes | yellow }})
 `
 
+type stageRequest struct {
+	ctx        context.Context
+	repository string
+	branch     string
+	params     *api.StageObjectParams
+	body       api.StageObjectJSONRequestBody
+}
+
+type stageResponse struct {
+	resp  *api.StageObjectResponse
+	error error
+}
+
+func stageWorker(wg *sync.WaitGroup, requests <-chan *stageRequest, responses chan<- *stageResponse) {
+	defer wg.Done()
+	client := getClient()
```

Comment:
Can we share clients?  I think HTTP connection pooling/reuse anyway happens at a level below, so it might manage to _reduce_ connection re-use which could _increase_ performance (or _decrease_ performance...).

Reply:
From a different angle - getClient should be called once, it can fail and we do not want to dump multiple errors.
They all share the same transport - but we don't need to perform the same operation multiple times to construct it.

Reply:
Done